pkg/install/4-addcustomisations.go

```diff
@@ -0,0 +1,116 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+func (i *Installer) addCustomisations(ctx context.Context) error {
+	restConfig, err := restconfig.RestConfig(ctx, i.env, i.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	{
+		cli, err := configclient.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		i.log.Print("waiting for version clusterversion")
+		timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Minute)
+		defer cancel()
+		err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+			cv, err := cli.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+			if err == nil {
+				for _, cond := range cv.Status.Conditions {
+					if cond.Type == configv1.OperatorAvailable && cond.Status == configv1.ConditionTrue {
+						return true, nil
+					}
+				}
+			}
+			return false, nil
+
+		}, timeoutCtx.Done())
+		if err != nil {
+			return err
+		}
+
+		i.log.Print("disabling updates")
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			cv, err := cli.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			cv.Spec.Upstream = ""
+			cv.Spec.Channel = ""
+
+			_, err = cli.ConfigV1().ClusterVersions().Update(cv)
+			return err
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	{
+		cli, err := operatorclient.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		var configGeneration int64
+		i.log.Print("updating console branding")
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			operatorConfig, err := cli.OperatorV1().Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			operatorConfig.Spec.Customization.Brand = operatorsv1.BrandAzure
+			configGeneration = operatorConfig.Status.ObservedGeneration
+
+			_, err = cli.OperatorV1().Consoles().Update(operatorConfig)
+			return err
+		})
+		if err != nil {
+			return err
+		}
+
+		i.log.Print("waiting for console to reload")
+		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+		defer cancel()
+		err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+			operatorConfig, err := cli.OperatorV1().Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+			if err == nil && operatorConfig.Status.ObservedGeneration > configGeneration {
```

Comment:
I'd expect to see you test operatorConfig.Status.ObservedGeneration == operatorConfig.Generation here - is there a particular reason why not to?